src/libprojectM/Renderer/MenuText.cpp

```diff
@@ -0,0 +1,163 @@
+#ifdef USE_TEXT_MENU
+
+#include "MenuText.h"
+
+#include "Common.hpp"
+
+#define GLT_IMPLEMENTATION
+#include "gltext.h"
+
+MenuText::MenuText(int viewportWidth)
+    : _viewportWidth(viewportWidth)
+{
+    gltInit();
+    _glTextInstance = gltCreateText();
+}
+
+MenuText::~MenuText()
+{
+    if (_glTextInstance)
+    {
+        gltDeleteText(_glTextInstance);
+        _glTextInstance = nullptr;
+    }
+}
+
+void MenuText::SetViewportWidth(int viewportWidth)
+{
+    _viewportWidth = viewportWidth;
+}
+
+void MenuText::CleanUp()
+{
+    gltTerminate();
+}
+
+void MenuText::DrawBegin() const
+{
+    // Begin text drawing (this for instance calls glUseProgram)
+    gltBeginDraw();
+}
+
+void MenuText::DrawEnd() const
+{
+    // Finish drawing text - will unbind font texture & shaders.
+    gltEndDraw();
+}
+
+void MenuText::Draw(std::string textLine,
+                    GLfloat x,
+                    GLfloat y,
+                    GLfloat scale,
+                    HorizontalAlignment horizontalAlignment,
+                    VerticalAlignment verticalAlignment,
+                    float r,
+                    float b,
+                    float g,
+                    float a,
+                    bool highlightable,
+                    const std::string& highlightText) const
+{
+    if (!gltInitialized || !_glTextInstance)
+    {
+        return;
+    }
+
+    int gltHorizontalAlignment = static_cast<int>(horizontalAlignment);
+    int gltVerticalAlignment = static_cast<int>(verticalAlignment);
+
+    gltSetText(_glTextInstance, textLine.c_str());
+    GLfloat textWidth = gltGetTextWidth(_glTextInstance, scale);
+
+    auto windowWidth = static_cast<float>(_viewportWidth);
+    if (gltHorizontalAlignment == GLT_LEFT)
+    {
+        // if left aligned factor in X offset
+        windowWidth -= x;
+    }
+
+    // if our window is greater than the text width, there is no overflow so let's display it normally.
+    if (windowWidth < textWidth)
+    {
+        // if the text is greater than the window width, we have a problem.
+        while (textWidth > windowWidth)
+        {
+            textLine.pop_back();
+            gltSetText(_glTextInstance, textLine.c_str());
+
+            gltDrawText2DAligned(_glTextInstance, x, y, scale, gltHorizontalAlignment, gltVerticalAlignment);
+            textWidth = gltGetTextWidth(_glTextInstance, scale);
+        }
+
+        // if it's not multi-line then append a ...
+        if (textLine.find('\n') != std::string::npos)
+        {
+           textLine.pop_back();
+           textLine.pop_back();
+           textLine.pop_back();
+           textLine += "...";
+        }
+    }
+
+    // redraw without transparency
```

Comment:
Not sure why it was drawn transparent Probably someone thought it's required to get the actual text width, but that is simply calculated from text & scale.